workflow/controller/controller.go

```diff
@@ -341,6 +341,30 @@ func (wfc *WorkflowController) createSynchronizationManager(ctx context.Context)
 		return strconv.Atoi(value)
 	}
 
+	getSemaphoreStrategy := func(lockKey string) (config.SemaphoreStrategy, error) {
+		lockName, err := sync.DecodeLockName(lockKey)
+		if err != nil {
+			return config.SemaphoreStrategyDefault, err
+		}
+		configMap, err := wfc.kubeclientset.CoreV1().ConfigMaps(lockName.Namespace).Get(ctx, lockName.ResourceName, metav1.GetOptions{})
+		if err != nil {
+			return config.SemaphoreStrategyDefault, err
+		}
+
+		value, found := configMap.Data[fmt.Sprintf("%s-strategy", lockName.Key)]
```

Comment:
Why do you need this "%s-strategy" string formatting? Where does this get populated?

In general I do think it would be nice to move out key generation into functions, it just helps to trace usage better when you can just use the LSP to find references.

Reply:
The user configures this in config map.

I'm not 100% sure I follow the suggestion. How would the user configure semaphore strategy with a function?

Thx!

Reply:
Thanks for your clarification, I meant refactor the string formatting out to a function, but if this is the only place it is going to be used, it is probably fine to remain as it is. A comment about it would be great though. 

Thanks

Reply:
Oh, thanks! I understand! I will add a comment. I appreciate the review @isubasinghe

Reply:
Thanks for doing that! No worries! I've just approved this